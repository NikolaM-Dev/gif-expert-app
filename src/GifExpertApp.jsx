import { useState } from 'react';

import AddCategory from './components/AddCategory';

const CATEGORIES = ['One Punch', 'Samurai X', 'Dragon ball'];

const GifExpertApp = () => {
  const [categories, setCategories] = useState(CATEGORIES);

  return (
    <>
      <h2>GifExpertApp</h2>
      <hr />
      <AddCategory />
      <ol>
        {categories.map((category) => (
          <li key={category}>{category}</li>
        ))}
      </ol>
    </>
  );
};

export default GifExpertApp;
