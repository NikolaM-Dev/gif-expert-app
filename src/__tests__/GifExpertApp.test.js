import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import GifExpertApp from 'GifExpertApp';

describe('Tests on <GifExpertApp />', () => {
  it('should show <GifExpertApp /> correctly', () => {
    const wrapper = shallow(<GifExpertApp />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should show a categories list', () => {
    const categories = ['One Punch', 'Dragon Ball'];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

    expect(wrapper.find('GifGrid')).toHaveLength(categories.length);
    expect(wrapper).toMatchSnapshot();
  });
});
