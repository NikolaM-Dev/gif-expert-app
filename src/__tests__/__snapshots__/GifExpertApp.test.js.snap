// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifExpertApp /> should show <GifExpertApp /> correctly 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;

exports[`Tests on <GifExpertApp /> should show a categories list 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
